go/pkg/ddlog/ddlog.go

```diff
@@ -32,6 +32,18 @@ static int ddlogTransactionCommitDumpChanges(ddlog_prog hprog, uintptr_t arg) {
     return ddlog_transaction_commit_dump_changes(hprog, dumpChangesCb, arg);
 }
 
+extern void handleOutRecordArray(uintptr_t progIdx, ddlog_record_update *changes, size_t num_changes);
+
+static int ddlogTransactionCommitDumpChangesAsArray(ddlog_prog hprog, uintptr_t progIdx) {
+    ddlog_record_update *changes;
+    size_t num_changes;
+    int rc = ddlog_transaction_commit_dump_changes_as_array(hprog, &changes, &num_changes);
+    if (rc != 0 || num_changes == 0) return rc;
```

Comment:
@ryzhyk I am assuming that there is no need to call `ddlog_free_record_updates` in that case

Reply:
correct